SecretManagement.LastPass.Extension/SecretManagement.LastPass.Extension.psm1

```diff
@@ -79,46 +116,66 @@ function Set-Secret
         [Parameter(ValueFromPipelineByPropertyName)]
         [hashtable] $AdditionalParameters
     )
-    if($Secret -is [string]) {
-        $Secret = @{
-            URL = "http://sn"
-            Notes = $Secret
-        }
-    }
-
     $sb = [System.Text.StringBuilder]::new()
-    if($Secret.UserName) {
-        $sb.Append("Username: ").AppendLine($Secret.UserName)
-        $sb.Append("login: ").AppendLine($Secret.UserName)
+    
+    
+    if ($Secret -is [string]) {
+        $Secret = @{Notes = $Secret}
+    } elseif ($Secret -is [pscredential]) {
+        $Secret = @{Username = $Secret.Username; Password = $Secret.GetNetworkCredential().password}
     }
 
-    if($Secret.Password) {
-        $pass = $Secret.Password
-        if ($Secret -is [pscredential]) {
-            $pass = $Secret.GetNetworkCredential().password
-        } elseif ($pass -is [securestring]) {
-            $pass = $pass | ConvertFrom-SecureString
-        }
 
-        $sb.Append("Password: ").AppendLine($pass)
-        $sb.Append("password: ").AppendLine($pass)
-        $sb.Append("sudo_password: ").AppendLine($pass)
-    }
-
-    if($Secret.URL) {
-        $sb.Append("URL: ").AppendLine($Secret.URL)
-    }
+    
+    if ($Secret -is [hashtable]){
+        $SpecialKeys = @('Language', 'NoteType', 'Notes')
+        $Keys = $Secret.Keys.Where({$_ -notin $SpecialKeys })
 
-    if($Secret.Notes) {
-        $sb.AppendLine("Notes:").AppendLine($Secret.Notes)
-    }
+        foreach ($k in $Keys) {
+            if ($Secret.$k -is [securestring]) {
+                $BSTR = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($Secret.$k)
+                $Secret.$k = [System.Runtime.InteropServices.Marshal]::PtrToStringAuto($BSTR)
+            }
+            [Void]($sb.AppendLine("$($k): $($Secret.$k)"))
+        }
 
+        # Notes need to be on a new line
+        if ($null -ne $Secret.Notes) {
+            if ($Secret.Notes -is [securestring]) {
+                $BSTR = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($Secret.Notes)
+                $Secret.Notes = [System.Runtime.InteropServices.Marshal]::PtrToStringAuto($BSTR)
+            }
+            [Void]($sb.AppendLine("Notes: `n$($Secret.Notes)"))
+        }
+    } 
+    
     try {
-        $sb.ToString() | Invoke-lpass 'add', $Name, '--non-interactive'
-    } catch {
+        $res = Invoke-lpass 'show', '--sync=now', '--name', $Name -ErrorAction SilentlyContinue
```

Comment:
This one I really do not understand...  When adding a secret and doing a "Show", it return the error  Could not find specified account(s). This would be expected if I did not put -ErrorAction SilentlyContinue.

Reply:
Redirecting the error stream through `someExe --argument 2>$null`
https://stackoverflow.com/questions/35940593/suppressing-powershell-errors-when-calling-external-executables